tests/manage/z_cluster/test_must_gather.py

```diff
@@ -51,3 +55,44 @@ def test_must_gather(self, mustgather, log_type):
         """
         mustgather.log_type = log_type
         mustgather.validate_must_gather()
+
+    @tier1
+    @bugzilla("1884546")
+    def test_must_gather_network_failure(self, nodes):
+        """
+        Test Must Gather Network Failure
+        """
+        logging.info("Create must_gather_thread and network_failure_thread")
+        mustgather = MustGather()
+        must_gather_thread = threading.Thread(target=mustgather.collect_must_gather)
+        network_failure_thread = threading.Thread(
+            target=self.disconnect_worker_nodes, nodes=nodes
+        )
+
+        logging.info("Starting must gather thread")
+        must_gather_thread.start()
+
+        logging.info("Sleep 15 Sec")
+        time.sleep(15)
+
+        logging.info("Starting network failure thread")
+        network_failure_thread.start()
+
+        logging.info("wait until network_failure thread is completely executed")
+        network_failure_thread.join()
+
+        logging.info("wait until must_gather thread is completely executed")
+        must_gather_thread.join()
+
+        mustgather.log_type = "OTHERS"
+        mustgather.validate_must_gather()
+
+    def disconnect_worker_nodes(self, nodes):
+        """
+        Disconnect Worker Nodes
+
+        """
+        worker_node_names = node.get_worker_nodes()
+        node.node_network_failure(node_names=worker_node_names, wait=True)
+        logger.info(f"Stop and start the worker nodes: {worker_node_names}")
+        nodes.restart_nodes_by_stop_and_start(node.get_node_objs(worker_node_names))
```

Comment:
We need a teardown step here. because if the test fail node will not come back online

Reply:
fixed